### ISSUE TITLE
False positive error for call expression that uses a callback protocol with a ParamSpec

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3417]
The following sample generates a false positive error:
```python
import contextlib
from concurrent.futures import Future, ThreadPoolExecutor
from typing import Callable, Iterator, ParamSpec, Protocol, TypeVar

P = ParamSpec("P")
T = TypeVar("T")

class TakesFunctionWithArguments(Protocol):
    def __call__(
        self, func: Callable[P, T], *args: P.args, **kwargs: P.kwargs
    ) -> Future[T]:
        ...

@contextlib.contextmanager
def submit_wrapper() -> Iterator[TakesFunctionWithArguments]:
    with ThreadPoolExecutor() as pool:

        def my_submit(
            func: Callable[P, T], *args: P.args, **kwargs: P.kwargs
        ) -> Future[T]:
            return pool.submit(func, *args, **kwargs)

        yield my_submit

def foo(a: int, b: int, c: int) -> int:
    return a + b + c

with submit_wrapper() as submit:
    submit(foo, a=1, b=2, c=3)
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.244, which I just published. It will also be included in the next release of pylance.
